core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -259,99 +250,113 @@ oppia.controller('StateResponses', [
             var interactionId = stateInteractionIdService.savedMemento;
 
-            var rulesServiceName =
+            var rulesServiceName = (
               AngularNameService.getNameOfInteractionRulesService(
-                interactionId);
+                interactionId));
 
             // Inject RulesService dynamically.
             var rulesService = $injector.get(rulesServiceName);
 
-            // See the training panel directive in StateEditor for an
-            // explanation on the structure of this object.
-            $scope.classification = {
-              answerGroupIndex: 0,
-              newOutcome: null
+            var fetchAndShowUnresolvedAnswers = function(expId, stateName) {
+              var unresolvedAnswersUrl = UrlInterpolationService.interpolateUrl(
+                '/createhandler/get_top_unresolved_answers/<exploration_id>', {
+                  exploration_id: expId
+                });
+              $http.get(unresolvedAnswersUrl, {
+                params: {
+                  state_name: _stateName
+                }
+              }).success(function(response) {
+                $scope.showUnresolvedAnswers(response.unresolved_answers);
+              }).error(function(response) {
+                $scope.showUnresolvedAnswers([]);
+              });
             };
 
-            FocusManagerService.setFocus('testInteractionInput');
+            $scope.showUnresolvedAnswers = function(unresolvedAnswers) {
+              $scope.showLoadingDots = false;
+              $scope.unresolvedAnswers = [];
+
+              unresolvedAnswers.forEach(function(item) {
+                var answer = item.answer;
+                var answerTemplate = (
+                  ExplorationHtmlFormatterService.getAnswerHtml(
+                    answer, stateInteractionIdService.savedMemento,
+                    stateCustomizationArgsService.savedMemento));
+                var classificationResult = (
+                  AnswerClassificationService.getMatchingClassificationResult(
+                    _explorationId, _stateName, _state, answer, rulesService));
+                var feedbackHtml = (
+                  classificationResult.outcome.feedback.getHtml());
+                var classificationType = (
+                  classificationResult.classificationCategorization);
+                if (classificationType !== EXPLICIT_CLASSIFICATION &&
+                  classificationType !== TRAINING_DATA_CLASSIFICATION &&
+                  !TrainingDataService.isConfirmedUnclassifiedAnswer(answer)) {
+                  $scope.unresolvedAnswers.push({
+                    answer: answer,
+                    answerTemplate: answerTemplate,
+                    classificationResult: classificationResult,
+                    feedbackHtml: feedbackHtml
+                  });
+                }
+              });
+            };
+
+            $scope.confirmToTrainingData = function(answerIndex) {
```

Comment:
I'm not sure what this function does based on its name. Could you elaborate? What does "confirm to" mean?